webpack/assets/javascripts/react_app/components/charts/StatisticsChartsList.js

```diff
@@ -1,111 +1,49 @@
-import React, {PropTypes} from 'react';
-import helpers from '../../common/helpers';
+import React, { PropTypes } from 'react';
 import chartService from '../../../services/statisticsChartService';
 import ChartBox from './ChartBox';
-import { STATUS } from '../../constants';
-import StatisticsStore from '../../stores/StatisticsStore';
-import StatisticsChartActions from '../../actions/StatisticsChartActions';
 import './StatisticsChartsListStyles.css';
+import { connect } from 'react-redux';
+import * as StatisticsChartActions from '../../redux/actions/statistics';
+import { STATUS } from '../../constants';
 
-class StatisticsChartsList extends React.Component {
-  constructor(props) {
-    super(props);
-
-    this.state = {charts: this.stateSetup(this.props.data)};
-
-    helpers.bindMethods(this, [
-      'onChange',
-      'onError']
-    );
+const getStatusFromChart = (chart) => {
```

Comment:
Based on what you're saying, the `Loader` would not be necessary but I'm not sure about the `ChartBox`. 

Generally, what thoughts about the pros and cons of using components like the [`Loader`](https://github.com/theforeman/foreman/blob/develop/webpack/assets/javascripts/react_app/components/common/Loader.js) or [`Fade`](https://github.com/gailsteiger/foreman/blob/pf-react-toast/webpack/assets/javascripts/react_app/components/common/Fade.js) which wrap other components and manage part of the wrapped components' rendering logic as opposed to including the relevant logic in the wrapped components themselves?